src/components/PeopleList/PeopleList.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useContext } from 'react';
+import { PeopleContext } from '../../PeopleContext';
+import { PeopleItem } from '../PeopleItem/PeopleItem';
+import { useLocation} from 'react-router-dom';
+
+export const PeopleList: React.FC = () => {
+  const { persons } = useContext(PeopleContext);
+  const location = useLocation();
+  const searchParams = new URLSearchParams(location.search);
+
+  const sex = searchParams.get('sex');
+  const query = searchParams.get('query');
+
+  const filterByGender = () => {
+    if (sex === 'm') {
+      return persons.filter((person) => person.sex === 'm');
+    }
+
+    if (sex === 'f') {
+      return persons.filter((person) => person.sex === 'f');
+    }
+
+    return persons;
+  };
+
+  const filteredPeople = () => {
+    if (query?.trim()) {
+      return persons.filter((person) => {
+        return person.name.toLowerCase().includes(query.toLowerCase());
+      });
+    }
+
+    if (sex !== 'all') {
+      return filterByGender();
+    }
+
+    // if (sex && sex !== 'all' && query && query.trim()) {
+    //   return filterByGender().filter((person) => {
+    //     return person.name.toLowerCase().includes(query.toLowerCase());
+    //   });
+    // }
+
+    return persons;
+  };
+
+  return (
+    <tbody>
+      {filteredPeople().map((person) => (
```

Comment:
it seems like you don't filter it by centuries:
![image](https://github.com/mate-academy/react_people-table-advanced/assets/86923089/b878e520-bfd9-43cb-bb78-46c0e6eea40d)

![image](https://github.com/mate-academy/react_people-table-advanced/assets/86923089/d5723a10-aec7-47e9-ae13-db98f5f85734)

also, note that filters should work together, for example, you should take into account all filters - filter by gender if selected and filter the result by centuries if any are selected. Also, selecting some person (when clicking by name) shouldn't reset all filters